src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/notifications/CaseNotesNotificationsService.kt

```diff
@@ -0,0 +1,88 @@
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.CaseNoteEventType
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.exception.AsyncEventExceptionHandling
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.CaseNote
+
+@Service
+class CaseNotesNotificationsService(
+  @Value("\${notify.templates.case-note-sent}") private val sentTemplate: String,
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.service-provider.case-note-details}") private val spCaseNoteLocation: String,
+  @Value("\${interventions-ui.locations.probation-practitioner.case-note-details}") private val ppCaseNoteLocation: String,
+  private val emailSender: EmailSender,
+  private val referralService: ReferralService,
+  private val hmppsAuthService: HMPPSAuthService,
+  private val userTypeChecker: UserTypeChecker,
+  private val communityAPIOffenderService: CommunityAPIOffenderService,
+) : ApplicationListener<CaseNoteEvent>, NotifyService {
+  companion object : KLogging()
+
+  @AsyncEventExceptionHandling
+  override fun onApplicationEvent(event: CaseNoteEvent) {
+    when (event.type) {
+      CaseNoteEventType.SENT -> {
+        emailAssignedCaseWorker(event.caseNote)
+        emailResponsibleProbationPractitioner(event.caseNote)
+      }
+    }
+  }
+
+  private fun emailResponsibleProbationPractitioner(caseNote: CaseNote) {
+    val sender = caseNote.sentBy
+    val responsibleOfficer = referralService.getResponsibleProbationPractitioner(caseNote.referral)
+    val senderIsResponsibleOfficer =
+      userTypeChecker.isProbationPractitionerUser(sender) && (
+        (responsibleOfficer.authUser?.let { it == sender } ?: false) ||
+          (responsibleOfficer.staffId?.let { it == communityAPIOffenderService.getStaffIdentifier(sender) }
+            ?: false) // if the RO doesn't have a staff ID we cannot determine if they are the sender, so assume not
+        )
+
+    // if the case note was sent by someone other than the RO, email the RO
+    if (!senderIsResponsibleOfficer) {
+      emailSender.sendEmail(
+        sentTemplate,
+        responsibleOfficer.email,
+        mapOf(
+          "recipientFirstName" to responsibleOfficer.firstName,
+          "referralReference" to caseNote.referral.referenceNumber!!,
+          "caseNoteUrl" to generateResourceUrl(interventionsUIBaseURL, ppCaseNoteLocation, caseNote.id).toString(),
+        )
+      )
+    }
+  }
+
+  private fun emailAssignedCaseWorker(caseNote: CaseNote) {
+    val sender = caseNote.sentBy
+    val assignee = caseNote.referral.currentAssignee
+    val senderIsAssignee = sender == assignee
+
+      // if the case note was sent by someone other than the assignee, email the assignee
+    if (!senderIsAssignee) {
+      val assigneeDetails = hmppsAuthService.getUserDetail(
+        assignee
+          ?: run {
```

Comment:
assignee will never be null here

reason is `sender` is not nullable. and this block is gated on `sender == assignee`